### ISSUE TITLE
file module state=directory fail when directory already exist

### ISSUE THREAD

**Issue by jwfang** [ansible/ansible-modules-core#555]
i just want to make sure the directory exist.
and there is no way to distinguish the "already exist" error

**Comment by sysadmin75** [comment]
I'm unable to reproduce this issue. I'm using devel branch. This is the playbook:
- hosts: all
  gather_facts: no
  tasks:
  - file: path=/tmp/dir state=directory

This is the output:

# ansible-playbook symlink.yml --limit apache

PLAY [all] *******************************************************************\* 

TASK: [file path=/tmp/dir state=directory] ***********************************\* 
ok: [apache]

PLAY RECAP *******************************************************************\* 
apache                     : ok=1    changed=0    unreachable=0    failed=0

**Comment by abadger** [comment]
@jwfang If you could provide us more information to reproduce (a simple playbook, ansible version and platform, for instance) and the output that you're getting we could look into this further.  Otherwise I'll close this in a few days since we can't reproduce it.

**Comment by benediktarnold** [comment]
I am seeing the same behaviour as @jwfang wit ansible 1.8.2
Will provide a simple usecase to reproduce this.

**Comment by benediktarnold** [comment]
That's a hard one. Seems to be some kind of race condition. Let me try to explain:
Some information about my setup:
localhost is running OSX 10.10
python version: 2.7.9
ansible version: 1.8.2
Repository containing inventory and playbook: https://github.com/benediktarnold/ansible-debug
Inventory:

```
server1
server2
```

Playbook:

```
- hosts: all 
  gather_facts: no 
  vars:
    base_dir: base
  tasks:
    - file: path=/tmp/dir/{{ base_dir }}/{{inventory_hostname}} state=directory
```

Run the playbook locally and make sure /tmp/dir does not exist on your localhost before you run it:

```
ansible-playbook -i inventory file-exists.yml --connection=local
```

Now the results:
Sometimes I get this error:

```
PLAY [all] ********************************************************************

TASK: [file path=/tmp/dir/{{ base_dir }}/{{inventory_hostname}} state=directory] ***
failed: [server2] => {"failed": true, "parsed": false}
Traceback (most recent call last):
  File "/Users/ben/.ansible/tmp/ansible-tmp-1422431465.35-122262796637628/file", line 1924, in <module>
    main()
  File "/Users/ben/.ansible/tmp/ansible-tmp-1422431465.35-122262796637628/file", line 238, in main
    os.mkdir(curpath)
OSError: [Errno 17] File exists: '/tmp/dir/base'

changed: [server1]

PLAY RECAP ********************************************************************
           to retry, use: --limit @/Users/ben/file-exists.retry

server1                    : ok=1    changed=1    unreachable=0    failed=0
server2                    : ok=0    changed=0    unreachable=0    failed=1
```

But most of the time everything works as expected:

```
PLAY [all] ********************************************************************

TASK: [file path=/tmp/dir/{{ base_dir }}/{{inventory_hostname}} state=directory] ***
changed: [server2]
changed: [server1]

PLAY RECAP ********************************************************************
server1                    : ok=1    changed=1    unreachable=0    failed=0
server2                    : ok=1    changed=1    unreachable=0    failed=0
```

I expect the following directory layout after I run the playbook:

```
{9:01}[1.9.3]/tmp ➭ tree dir
dir
└── base
    ├── server1
    └── server2

3 directories, 0 files
```

**Comment by sean-abbott** [comment]
We're running into the same issue on a directory that is used by gluster.  If the directory exists, then we get:  

TASK: [gluster_client | Make gluster directory] *******************************
failed: [10.7.30.200] =>
{"failed": true, "parsed": false}

SUDO-SUCCESS-wsywsozyaxukezjpcsifqjgfisekrcax
Traceback (most recent call last):
File "/home/user/.ansible/tmp/ansible-tmp-1440101900.06-193442105882759/file", line 1924, in <module>
main()
File "/home/user/.ansible/tmp/ansible-tmp-1440101900.06-193442105882759/file", line 238, in main
os.mkdir(curpath)
OSError: [Errno 17] File exists: '/mnt/hdfs'

The calling playbook is very simple:  

```
 - name: "Make gluster directory"
   sudo: yes 
   file:
    path={{ gluster_client_mountpoint }}
    state=directory
    owner={{ builder_user }}
    group={{ builder_group }} 
```

**Comment by gturri** [comment]
Just in case someone else stumble upon this bug, and consider spending time to fix it: it has already been fixed by #1649 

BTW, in my case, I could workaround this bug by explicitly limiting it to run only once:

```
  - file: path={{ my_file }} state=directory
    run_once: true
```

**Comment by Jmainguy** [comment]
@jwfang @abadger I think this can be closed, as #1649 solves this.

**Comment by Jmainguy** [comment]
@abadger @bcoca @jwfang Can this be closed, or should it remain open? See #1649

**Comment by bcoca** [comment]
closed via #1649
